### PR TITLE
feat(medusa-payment-stripe): Add delay to Stripe webhook

### PR DESCRIPTION
**Pull request by olivermrbl**
**What**
- Migrate Stripe plugin to use API Routes + Subscribers
- Change the behaviour of Stripe webhook
Instead of processing webhook events immediately, we fire an event to perform the work in the background. The primary reason for changing the webhook is that the previous implementation led to occasional conflicts due to a race condition between the client's and the webhook's request to complete a cart. Now, we emit an event with a configurable delay (defaults to 2s) to prevent this from happening

This approach was preferred over adding a timeout directly to the webhook execution, as it is generally best practice to respond to a webhook event immediately after receiving it.

### CODE REVIEW

**Review by adrien2p — APPROVED** [review]
Looks very clean, we need to test it :)

**Comment by olivermrbl** [comment]
@adrien2p, have tested this locally on my machine :)

**Comment by adrien2p** [comment]
Splendide, you ve used the stripe cli?

**Comment by olivermrbl** [comment]
Used their console + ngrok

**Comment by adrien2p** [comment]
Ok works too, eaiser with the cli 😅

**Comment by olivermrbl** [comment]
I see, felt like it was pretty easy with the console. I'll have to try the CLI next time

**Comment by adrien2p** [comment]
Should we move forward with this one?

**Comment by olivermrbl** [comment]
Yes, let me get it up to speed so we can merge :)

**Comment by olivermrbl** [comment]
@carlos-r-l-rodrigues, would you like to give this a review?

File: packages/medusa-payment-stripe/src/api/stripe/hooks/route.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { MedusaRequest, MedusaResponse } from "@medusajs/medusa"
+import { constructWebhook } from "../../utils/utils"
+
+const WEBHOOK_DELAY = process.env.STRIPE_WEBHOOK_DELAY ?? 5000 // 5s
+const WEBHOOK_RETRIES = process.env.STRIPE_WEBHOOK_RETRIES ?? 3
+
+export const POST = async (req: MedusaRequest, res: MedusaResponse) => {
+  let event
```
Comment (carlos-r-l-rodrigues) [inline packages/medusa-payment-stripe/src/api/stripe/hooks/route.ts]: nit: `let` can be a `const` inside `try`

**Review by carlos-r-l-rodrigues — APPROVED** [review]
LGTM

**Comment by piereligio** [comment]
Hey there, a question: would it be possible/beneficial to update medusa-payment-paypal plugin in the same way? I think it might fix some issues I'm meeting on it, but I am not sure about it. Thanks

**Comment by olivermrbl** [comment]
@piereligio, I'll look into how much effort it would require. 

PayPal can be cumbersome to work with directly through their SDKs. Are you aware, that you can use PayPal via Stripe (which is also supported by this plugin)?

**Comment by piereligio** [comment]
> @piereligio, I'll look into how much effort it would require. 
> 
> PayPal can be cumbersome to work with directly through their SDKs. Are you aware, that you can use PayPal via Stripe (which is also supported by this plugin)? 

Oh, I had no idea! Has it to be enabled manually somehow? Thanks for the info!

**Comment by piereligio** [comment]
> @piereligio, I'll look into how much effort it would require.
> 
> PayPal can be cumbersome to work with directly through their SDKs. Are you aware, that you can use PayPal via Stripe (which is also supported by this plugin)?

Sorry for the second reply. I asked stripe to add paypal to my account and it's pending. In while I'm trying on developer mode, where it's enabled already. It should just pop as an option? I don't see it. I am on nextJS medusa official storefront, I see only credit card fields. Maybe I'm using wrong frontend stripe plugins or I missed some configuration to do? Thanks in advance

**Comment by piereligio** [comment]
> @piereligio, I'll look into how much effort it would require.
> 
> PayPal can be cumbersome to work with directly through their SDKs. Are you aware, that you can use PayPal via Stripe (which is also supported by this plugin)?

Oh, I had no idea, thanks for the info!
I noticed the snapshot mentioned here got rid of all the webhook issues I got from stripe, so it's working very well.

But how can I add more payment methods in stripe? I enabled automatic_payment_methods from medusa-config.js, but nothing changed. Basically in the storefront I only see the credit card option, with the fields for it. For now I enabled Google Pay and Apple Pay (PayPal is pending). Do I need to write my own custom code for storefront?

**Pull request merged**